### PR TITLE
Add return keyword & exec() built-in function

### PR DESCRIPTION
**Pull request by sauerbraten**
This will allow returning a value from a template and using that value in the calling template, enabling users to have "function" templates they can reuse.

foo.jet:
```
{{ f := "f" }}
{{ o := "o" }}

... some content that will be discarded when this template runs inside exec() ...

{{ return f+o+o }}
```

test.jet:
```
{{ foo := exec("./foo") }}

<p>below the line, there will only be "foo":</p>
<hr>
{{ foo }}
```

### CODE REVIEW

**Comment by annismckenzie** [comment]
Uuuuh, this looks neat! Can you add some unit tests?

**Comment by sauerbraten** [comment]
Here you go!

File: testData/execReturn/foo.jet [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{{ f := "f" }}
+{{ o := "o" }}
+... some content that will be discarded when this template runs inside exec() ...
+{{ return f+o+o }}
```
Comment (annismckenzie) [inline testData/execReturn/foo.jet]: Nit: can you add a trailing newline?

File: testData/execReturn/test.jet [hunk 2]
```diff
@@ -0,0 +1 @@
+{{ foo := exec("./foo") }}{{ foo }}
```
Comment (annismckenzie) [inline testData/execReturn/test.jet]: In this file as well.

**Comment by annismckenzie** [comment]
After we merge this it'd be nice if you added a couple lines of documentation to the wiki for this feature. I'll arrange access for you.

File: testData/execReturn/foo.jet (hunk 1 above)
Comment (sauerbraten) [reply testData/execReturn/foo.jet]: 0085ed7 :+1:

File: testData/execReturn/test.jet (hunk 2 above)
Comment (sauerbraten) [reply testData/execReturn/test.jet]: 0085ed7 :+1:

File: testData/execReturn/foo.jet (hunk 1 above)
Comment (annismckenzie) [reply testData/execReturn/foo.jet]: And now your tests are failing because of the added newlines. I guess we don't want those, right?

Comment (sauerbraten) [reply testData/execReturn/foo.jet]: We don't want the newlines or we don't want the test failures? :D
Anyway, I somehow had to change only one of the tests... is this expected behavior?
db4d56e

**Comment by sauerbraten** [comment]
I'd love to, but I don't have permission to do that :D
No write access 🤷‍♂

**Comment by annismckenzie** [comment]
I think it's time – there you go. 😬

**Comment by sauerbraten** [comment]
Is there a policy whether to merge as-is or can/should I squash-merge?

**Comment by annismckenzie** [comment]
Always use `Rebase and merge`, please. 👍

**Comment by annismckenzie** [comment]
No squashing. I like keeping the history.

**Pull request merged**